tools/gen_version.sh

```diff
@@ -15,12 +15,12 @@
 
 FILE="$1"
 
-if [[ ! -z "$2" ]]; then
+if test -n "$2"; then
     # second arg passed when called from ocamlformat.opam
     VERSION="$2"
 else
     # second arg omitted when called from src/dune
-    if [[ ! "%%VERSION%%" == "%%"*"%%" ]]; then
+    if test "%%VERSION%%" != "%%"*"%%"; then
```

Comment:
Did you test this? It does not behave the same as before at least in the case when passed only one arg.

Reply:
You are right, didn't find a way to make this more portable though